apps/api-key-service/src/routes/keys/KeysRoute.ts

```diff
@@ -0,0 +1,82 @@
+import crypto from 'crypto'
+import type { Logger } from 'pino'
+import { z } from 'zod'
+
+import type { Database } from '@/db/Database'
+import type { ApiKey } from '@/models/apiKeys'
+import { createApiKey, getApiKey } from '@/models/apiKeys'
+import type { Metrics } from '@/monitoring/metrics'
+import { Route } from '@/routes/Route'
+import { Trpc } from '@/Trpc'
+
+export class KeysRoute extends Route {
+  public readonly name = 'keys' as const
+
+  public readonly createApiKey = 'createApiKey' as const
+  public readonly createApiKeyController = this.trpc.procedure
+    .input(
+      z.object({
+        entityId: z.string(),
+        state: z
+          .union([z.literal('enabled'), z.literal('disabled')])
+          .optional()
+          .default('disabled'),
+        key: z.string().optional().default(crypto.randomUUID()),
+      }),
+    )
+    .mutation(async ({ input }) => {
+      const { entityId, state, key } = input
+
+      let apiKey: ApiKey
+      try {
+        apiKey = await createApiKey(this.database, {
+          entityId,
+          state,
+          key,
+        })
+        this.metrics.generatedApiKeysCount.inc()
+      } catch (e) {
+        throw this.logAndReturnDbError(e, 'Error creating API key')
+      }
+
+      return { apiKey }
+    })
+
+  public readonly getApiKey = 'getApiKey' as const
+  public readonly getApiKeyController = this.trpc.procedure
+    .input(
+      z.object({
+        id: z.string(),
+      }),
+    )
+    .query(async ({ input }) => {
+      let apiKey: ApiKey | null
+      try {
+        apiKey = await getApiKey(this.database, input.id)
```

Comment:
this is an internal service. I was thinking that any auth logic should be implemented at the api gateway that utilize this service. api-key-service doesn't have any auth logic. 

`entityId` is a dapp-console-api specific business logic that we're only keeping in api-key-service for convenient record keeping. think it will be more maintainable if the auth logic is kept at the public api level.